model/src/order.rs

```diff
@@ -42,7 +43,7 @@ impl Order {
         order_creation: OrderCreation,
         domain: &DomainSeparator,
     ) -> Option<Self> {
-        let owner = order_creation.validate_signature(domain)?;
+        let owner = order_creation.validate_signature(domain, order_creation.signature)?;
```

Comment:
nit: I don't like that we are passing a value from the object itself into a method on this object (it should have access to `self.signature` inside `validate_signature` and there is no reason to pass any other signature).

I think we can expose a `validate_signature(domain)` on both `OrderCreation` and `OrderCancellation` which internally calls into the shared helper methods.

Reply:
Right, I could copy-paste the validate signature methods into these structs directly (but then we have duplicated code). Traits can't have fields and as a result I cannot access signature if the method is defined in the trait.